zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/cache/ProcessCache.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.cache;
+
+import com.github.benmanes.caffeine.cache.CacheLoader;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import io.camunda.search.entities.FlowNodeInstanceEntity;
+import io.camunda.search.entities.UserTaskEntity;
+import io.camunda.zeebe.gateway.rest.config.GatewayRestConfiguration;
+import io.camunda.zeebe.gateway.rest.util.XmlUtil;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Process cache uses a Caffeine {@link LoadingCache} to store process definition key and {@link
+ * ProcessCacheItem} entries.
+ *
+ * <p>Use the {@link ProcessCache#getCacheItem(long)} method to load one item or the {@link
+ * ProcessCache#getCacheItems(Set)} method to load multiple cache items at once.
+ *
+ * <p>The process cache default configuration can be changed via the {@link
+ * GatewayRestConfiguration.ProcessCacheConfiguration} properties.
+ */
+public class ProcessCache {
+
+  private final LoadingCache<Long, ProcessCacheItem> cache;
+  private final XmlUtil xmlUtil;
+
+  public ProcessCache(final GatewayRestConfiguration configuration, final XmlUtil xmlUtil) {
+    this.xmlUtil = xmlUtil;
+    final var cacheBuilder =
+        Caffeine.newBuilder().maximumSize(configuration.getProcessCache().getMaxSize());
+    final var ttlMillis = configuration.getProcessCache().getExpirationMillis();
+    if (ttlMillis != null && ttlMillis > 0) {
+      cacheBuilder.expireAfterAccess(ttlMillis, TimeUnit.MILLISECONDS);
```

Comment:
if a Cached item is actively read (via active polling from the client) , it can remain for a long time in the cache. Should we rather use `expireAfterWrite` ? 
a TTL is typically expressed on last write

Reply:
But if it's being read so frequently then it also makes sense to keep it in the cache, doesn't it?

Reply:
On a second though, I guess the problem is that it might remain in the cache for so long that it gets outdated. I'm not sure if that can happen with flow nodes but I get your point.
We can switch it to `expireAfterWrite`.

Reply:
I guess it will not be changed as we use processDefinitionKey as key of the cache map, and any change comes with new version of the process and a new key
well it is up to you :) you can keep expireAfterAccess and rename the configuration as `expirationIdleTime` instead of expiration

Reply:
Added the `idle` in the name.